### DISCUSSION TITLE
Trigger ValidateAll when Rowupdating error condition?

### DISCUSSION THREAD

**Discussion by sanguinal** [Megabit/Blazorise#2002] [Q&A]
Hi All,

Really finding the datagrid useful and powerful. But I may be missing the best way to update row to the database on the Editrow.

Currently I am using the RowUpdating function to commit the update to the database, there are on occasions server side conditions that one wants to communicate back to the fields as errors in the columns in the datagrid for the row being edited.

Currently the SaveWithValidation function issues a validations.ValidateAll() before Save.InvokeAsync and then the RowUpdating callback is called.

Is there a way to trigger a validateall from RowUpdating callback, to refresh the errors in the event of a e.Cancel = true condition?

I have captured the commandContext from save and issue a ctx.Clicked.InvokeAsync() which causes a validateAll when there are error conditions from the database, an example may be a unique username was used by another user at the exact same time. 

But I think I may be missing a better way? 

    private void SaveButton(CommandContext<ACCOUNT_DATASET_WITH_ACCESS> ctx)
    {
        saveContext = ctx;
        result = new ReturnStatus();
        ctx.Clicked.InvokeAsync();
    }

    async Task OnUpdating(CancellableRowChange<ACCOUNT_DATASET_WITH_ACCESS, Dictionary<string, object>> e)
    {
        result = new ReturnStatus();
        result.Status = sig_r_return.UpdateFail;

        if (e.Item.US_UPDATE)
        {
            result = await AccountData.UpdateAccount(e.Item, e.Values);
        }
        if (result.Status != sig_r_return.UpdateSuccess)
        {
            e.Cancel = true;
            bRowBeingEdited = true;
            saveContext?.Clicked.InvokeAsync();
        }
        else
        {
            bRowBeingEdited = false;
            saveContext = null;
        }

    }

Sanguinal

**Comment by stsrki** [comment]
You can have custom component, for example Save button with cascading validations

```cshtml
<Button Clicked="@Clicked" Color="Color.Primary">
    @ChildContent
</Button>

@code {
    [Parameter] public RenderFragment ChildContent { get; set; }

    [Parameter] public EventCallback Clicked { get; set; }

    [CascadingParameter] public Validations Validations { get; set; }
}
```

and

```cshtml
<SaveCommandTemplate>
    <MyButton @ref="@myButton" Clicked="@SaveButton">@context.LocalizationString</MyButton>
</SaveCommandTemplate>
```

and

```cs
myButton.Validations.ValidateAll()
```

**Reply by sanguinal** [reply]
That's awesome ill give that go and thanks for the fast response!

Sanguinal
